Fix CI

yvt opened the pull request:
This PR fixes CI failures that have occurred due to external reasons.

# Highlights

## `blessing`

Our `mach` script uses [`blessings`][2] for terminal rendering during WPT tests. The `setuptools` script of `blessings` 1.6 activates the option `use_2to3` to convert the code when installing the package for Python 3. However, this option has been [removed][1] in `setuptools` 58. This caused a breakage in our CI pipelines, producing an error message like the following:

```
Collecting blessings==1.6
  Downloading blessings-1.6.tar.gz (19 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/runner/work/servo/servo/python/_virtualenv3.8/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-n51ah1i5/blessings_f1c0f646d72e488e988232f7aeb42946/setup.py'"'"'; __file__='"'"'/tmp/pip-install-n51ah1i5/blessings_f1c0f646d72e488e988232f7aeb42946/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-1eg28bh4
         cwd: /tmp/pip-install-n51ah1i5/blessings_f1c0f646d72e488e988232f7aeb42946/
    Complete output (1 lines):
    error in blessings setup command: use_2to3 is invalid.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/af/4a/61acd1c6c29662d3fcbcaee5ba95c20b1d315c5a33534732b6d81e0dc8e8/blessings-1.6.tar.gz#sha256=edc5713061f10966048bf6b40d9a514b381e0ba849c64e034c4ef6c1847d3007 (from https://pypi.org/simple/blessings/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement blessings==1.6 (from versions: 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.5.1, 1.6, 1.6.1, 1.7)
ERROR: No matching distribution found for blessings==1.6
"/home/runner/work/servo/servo/python/_virtualenv3.8/bin/python" "-m" "pip" "install" "-I" "-r" "/home/runner/work/servo/servo/python/requirements.txt" failed with error code 1:
Output:
Error:
Error: Process completed with exit code 1.
```

Our `requirements.txt` pins the version of `setuptools`, but this didn't work probably because `virtualenv` uses an embedded version of `setuptools`, which was recently [upgraded][4] to 58 in `virtualenv` 20.8.0.

This PR upgrades `blessings` to a version that [doesn't use][3] the `use_2to3` option anymore to fix this issue.

## OpenSSL Binding

It appears that, when installing OpenSSL, Homebrew now installs version 3.0 by default. OpenSSL 3.0 isn't supported by `openssl-sys` 0.9.63 that we currently use and causes a build error while building the binding.

```
build/expando.c:10:1: error: pasting formed 'RUST_VERSION_OPENSSL_(', an invalid preprocessing token
VERSION(OPENSSL, OPENSSL_VERSION_NUMBER)
^
build/expando.c:5:23: note: expanded from macro 'VERSION'
#define VERSION(n, v) VERSION2(n, v)
                      ^
build/expando.c:4:43: note: expanded from macro 'VERSION2'
#define VERSION2(n, v) RUST_VERSION_##n##_##v
                                          ^
1 error generated.
```

This PR upgrades `openssl[-sys]` to newer versions that were [updated][5] to support OpenSSL 3.0.

## `clang-format`

`mach test-tidy` requires a specific version of `clang-format` so that the formatting result is consistent between build environments. [The virtual environment][6] used by GitHub Actions has been [updated][7] and now provides a version of `clang-format` that is different from the one expected by our `mach` tool. This PR updates our tool to address the issue.

[1]: https://github.com/pypa/setuptools/blob/main/CHANGES.rst#v5800
[2]: https://github.com/erikrose/blessings
[3]: https://github.com/erikrose/blessings/pull/132
[4]: https://github.com/pypa/virtualenv/blob/main/docs/changelog.rst#v2080-2021-09-16
[5]: https://github.com/sfackler/rust-openssl/commit/b4c735e8af546dee49b54679f63bcfec16c15a0f
[6]: https://github.com/actions/virtual-environments
[7]: https://github.com/actions/virtual-environments/pull/3905/files#diff-816219d3c9f30fcc1c40f88ca826c1662bf99eded781c1b419dd2dfb577ae227R380

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

---
- [ ] There are tests for these changes OR
- [x] These changes do not require tests because they don't change the production code

yvt committed:
chore(mach): upgrade `blessings` to 1.7

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/requirements.txt

rygo6 commented:
Just ran into this issue myself building on windows and this fixed it.

CYBAI reviewed (approved):
Thank you!

CYBAI commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 3d31ee5594a4abb17d3e1026711a74301d3480a8 with merge 63182288258a7d8884cd897c1c015786926d40ef...

bors-servo commented:
:broken_heart: Test failed - [checks-github](https://github.com/servo/servo/runs/3764604008)

yvt commented:
Build (Window UWP x64) is failing due to an unsupported version of `clang-format`:

```
clang-format version 12.0.1
clang-format: wrong version (v11 required). Skipping CPP formatting.
Error: can't find suitable clang-format version. Required with --force-cpp.
```

The build error in Build (macOS) points to code that was changed in <https://github.com/sfackler/rust-openssl/commit/b4c735e8af546dee49b54679f63bcfec16c15a0f>. I suspect our `openssl-sys` is too old to support OpenSSL 3.0.

<details>
<summary>Build (macOS), failed build log</summary>

```
The following warnings were emitted during compilation:

warning: build/expando.c:10:1: error: pasting formed 'RUST_VERSION_OPENSSL_(', an invalid preprocessing token
warning: VERSION(OPENSSL, OPENSSL_VERSION_NUMBER)
warning: ^
warning: build/expando.c:5:23: note: expanded from macro 'VERSION'
warning: #define VERSION(n, v) VERSION2(n, v)
warning:                       ^
warning: build/expando.c:4:43: note: expanded from macro 'VERSION2'
warning: #define VERSION2(n, v) RUST_VERSION_##n##_##v
warning:                                           ^
warning: 1 error generated.

error: failed to run custom build command for `openssl-sys v0.9.63`

Caused by:
  process didn't exit successfully: `/Users/runner/work/servo/servo/target/release/build/openssl-sys-c4bfcb0ae9ffc1fd/build-script-main` (exit status: 101)
  --- stdout
  cargo:rustc-cfg=const_fn
  cargo:rerun-if-env-changed=X86_64_APPLE_DARWIN_OPENSSL_LIB_DIR
  X86_64_APPLE_DARWIN_OPENSSL_LIB_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_LIB_DIR
  OPENSSL_LIB_DIR = /usr/local/opt/openssl@3/lib
  cargo:rerun-if-env-changed=X86_64_APPLE_DARWIN_OPENSSL_INCLUDE_DIR
  X86_64_APPLE_DARWIN_OPENSSL_INCLUDE_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_INCLUDE_DIR
  OPENSSL_INCLUDE_DIR = /usr/local/opt/openssl@3/include
  cargo:rustc-link-search=native=/usr/local/opt/openssl@3/lib
  cargo:include=/usr/local/opt/openssl@3/include
  OPT_LEVEL = Some("3")
  TARGET = Some("x86_64-apple-darwin")
  HOST = Some("x86_64-apple-darwin")
  CC_x86_64-apple-darwin = None
  CC_x86_64_apple_darwin = None
  HOST_CC = None
  CC = Some("clang")
  CFLAGS_x86_64-apple-darwin = None
  CFLAGS_x86_64_apple_darwin = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  running: "clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "/usr/local/opt/openssl@3/include" "-Wall" "-Wextra" "-E" "build/expando.c"
  cargo:warning=build/expando.c:10:1: error: pasting formed 'RUST_VERSION_OPENSSL_(', an invalid preprocessing token
  cargo:warning=VERSION(OPENSSL, OPENSSL_VERSION_NUMBER)
  cargo:warning=^
  cargo:warning=build/expando.c:5:23: note: expanded from macro 'VERSION'
  cargo:warning=#define VERSION(n, v) VERSION2(n, v)
  cargo:warning=                      ^
  cargo:warning=build/expando.c:4:43: note: expanded from macro 'VERSION2'
  cargo:warning=#define VERSION2(n, v) RUST_VERSION_##n##_##v
  cargo:warning=                                          ^
  cargo:warning=1 error generated.
  exit status: 1

  --- stderr
  thread 'main' panicked at '
  Header expansion error:
  Error { kind: ToolExecError, message: "Command \"clang\" \"-O3\" \"-ffunction-sections\" \"-fdata-sections\" \"-fPIC\" \"--target=x86_64-apple-darwin\" \"-I\" \"/usr/local/opt/openssl@3/include\" \"-Wall\" \"-Wextra\" \"-E\" \"build/expando.c\" with args \"clang\" did not execute successfully (status code exit status: 1)." }

  Failed to find OpenSSL development headers.

  You can try fixing this setting the `OPENSSL_DIR` environment variable
  pointing to your OpenSSL installation or installing OpenSSL headers package
  specific to your distribution:

      # On Ubuntu
      sudo apt-get install libssl-dev
      # On Arch Linux
      sudo pacman -S openssl
      # On Fedora
      sudo dnf install openssl-devel

  See rust-openssl README for more information:

      https://github.com/sfackler/rust-openssl#linux
  ', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.9.63/build/main.rs:147:13
  stack backtrace:
     0: rust_begin_unwind
               at /rustc/0fa3190394475a84360b34e074e719d519bc40f1/library/std/src/panicking.rs:517:5
     1: std::panicking::begin_panic_fmt
               at /rustc/0fa3190394475a84360b34e074e719d519bc40f1/library/std/src/panicking.rs:460:5
     2: build_script_main::validate_headers
     3: build_script_main::main
     4: core::ops::function::FnOnce::call_once
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
warning: build failed, waiting for other jobs to finish...
```

</details>

yvt committed:
chore(deps): bump `openssl-sys` from 0.9.63 to 0.9.67

The macOS CI now installs OpenSSL 3.0, which is only supported by newer
versions of `openssl-sys`.

yvt committed:
chore(mach): update the required clang-format version to 12

yvt committed:
chore(deps): bump `openssl` from 0.10.34 to v0.10.36

`openssl-sys` 0.9.67 seems incompatible with an older `openssl`.

yvt commented:
I think I'm going to expand the scope of this PR.

yvt commented:
@bors-servo try

bors-servo commented:
@yvt: :key: Insufficient privileges: not in try users

bors-servo commented:
:hourglass: Trying commit f375f3c4ef34005e298afaceaadf88b51aa05901 with merge 604fe56b42ba97f951d2d6fd1c48ce1033e4a766...

bors-servo commented:
:sunny: Test successful - [checks-github](https://github.com/servo/servo/runs/3793037713), [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/XhIbl0mrSZCegqRdBNPhaA)
State: approved= try=True

jdm commented:
@bors-servo r+
Thank you!

bors-servo commented:
:pushpin: Commit f375f3c has been approved by `jdm`

<!-- @bors-servo r=jdm f375f3c4ef34005e298afaceaadf88b51aa05901 -->

bors-servo commented:
:hourglass: Testing commit f375f3c4ef34005e298afaceaadf88b51aa05901 with merge 0e0e69788cbb826194f6b1b6d037897d282c5e80...

bors-servo commented:
:sunny: Test successful - [checks-github](https://github.com/servo/servo/runs/3799116571), [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/AtDzSXyASEqYyeBSO5QJxg)
Approved by: jdm
Pushing 0e0e69788cbb826194f6b1b6d037897d282c5e80 to master...
